### DISCUSSION TITLE
Controller validate running on other fields?

### DISCUSSION THREAD

**Discussion by a-rattana** [react-hook-form/react-hook-form#9050] [Q&A] [answered]
Hi I'm trying to figure out why editing one field is causing another field to rerun it's validation when using Controller, I have a working demo using inputs and register but I can't seem to get the same functionality with Controllers and custom components. Can someone please help?

Controller component rerunning validation on another field  
https://codesandbox.io/s/react-hook-form-v7-controller-forked-234izf?file=/src/index.js

Working fine when using html inputs and register
https://codesandbox.io/s/react-hook-form-async-submit-validation-forked-lfzu4l?file=/src/index.js

**Comment by leapful** [comment, marked as answer]
@a-rattana : Your 2 codesandbox urls are using different version of react-hook-form. If you upgrade react-hook-form to latest version for both, the behavior will be the same with validation on all fields. Note that this is an expected behavior for mode "onChange" because it will need to re-validate all fields in order to check form state valid or not if there is any dependency between field (for example: field A is not valid if field B date time is before today).
